source/server/configuration_impl.cc

```diff
@@ -141,6 +125,76 @@ void MainImpl::initializeTracers(const Json::Object& configuration, Instance& se
   }
 }
 
+void MainImpl::initializeStatsSinks(const Json::Object& configuration, Instance& server) {
+  ENVOY_LOG(info, "loading stats sink configuration");
+
+  // TODO(mrice32): DEPRECATED - statsd_local_udp_port, statsd_udp_ip_address,
+  // statsd_tcp_cluster_name fields in the base json configuration will be moved to subobjects
+  // inside the stats_sinks array for 1.5.0.
+  if (configuration.hasObject("statsd_local_udp_port") &&
+      configuration.hasObject("statsd_udp_ip_address")) {
+    throw EnvoyException("statsd_local_udp_port and statsd_udp_ip_address "
+                         "are mutually exclusive.");
+  }
+
+  std::vector<Json::ObjectSharedPtr> sinks = configuration.getObjectArray("stats_sinks", true);
+
+  // Used to convert the deprecated statsd configuration into a json snippet for consumption by the
+  // new sink config parsing logic.
+  const std::string statsd_json = R"EOF(
```

Comment:
See the `translate*()` conversion functions in `source/common/config/*_json.cc` for the stylized way we are mapping from v1 JSON to v2 proto.